src/actions/targetActions.js

```diff
@@ -1,19 +1,38 @@
-import { SubmissionError } from 'redux-form/immutable';
-
 import TargetApi from 'api/targetApi';
 import * as actions from './actionTypes';
 
 export const loadTargetsSuccess = targets => ({
   type: actions.LOAD_TARGETS_SUCCESS, targets
 });
 
+export const loadTargetsFailed = errors => ({
+  type: actions.LOAD_TARGETS_FAILED, errors
+});
+
+export const addTargetSuccess = target => ({
+  type: actions.ADD_TARGET_SUCCESS, target
+});
+
+export const addTargetFailed = errors => ({
+  type: actions.ADD_TARGET_FAILED, errors
+});
+
 export const loadTargets = () =>
   async (dispatch) => {
     try {
       await TargetApi.getTargets().then((data) => {
         dispatch(loadTargetsSuccess(data.targets));
       });
-    } catch (err) {
-      throw new SubmissionError(err.errors);
+    } catch (error) {
+      dispatch(addTargetFailed(error));
     }
-  };
\ No newline at end of file
+  };
+
+export const addTarget = target =>
+  async (dispatch) => {
+    await TargetApi.createTarget(target).then((data) => {
```

Comment:
If you use async/await you don't need to use `then` and `catch`. You can wait for the result on a variable assignment and wrap that in a `try-catch` clause, and then just go on with the code.
```suggestion
    try {
      const data = await TargetApi.createTarget(target);
      dispatch(addTargetSuccess(data.target));
    } catch(error) {
      ...
```

Reply:
I understand that promise has a own (catch)'s function in order to call this when is rejected. Is right?